addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1214,16 +1223,32 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
                 <we-button data-select-data-attribute="!fileSet">${_t("Is not set")}</we-button>
             </we-select>
         `)[0]);
+        const recursiveFindCircular = (el) => {
+            const dependencyEl = this._getDependencyEl(el);
+            if (!dependencyEl) {
+                return false;
+            }
+            if (dependencyEl.closest('.s_website_form_field').dataset.visibilityDependency === this._getFieldName()) {
+                return true;
+            } else {
+                return recursiveFindCircular(dependencyEl.closest('.s_website_form_field'));
+            }
+        };
 
         // Update available visibility dependencies
         const selectDependencyEl = uiFragment.querySelector('we-select[data-name="hidden_condition_opt"]');
         if (selectDependencyEl) {
             const existingDependencyNames = [];
-            for (const el of this.formEl.querySelectorAll('.s_website_form_field:not(.s_website_form_dnone)')) {
+            const currentFieldName = this._getFieldName();
+            // Exclude field that would create a direct circular dependency
+            const preventCircularSelector = `:not([data-visibility-dependency="${currentFieldName}"])`;
+            for (const el of this.formEl.querySelectorAll(`.s_website_form_field:not(.s_website_form_dnone)${preventCircularSelector}`)) {
                 const inputEl = el.querySelector('.s_website_form_input');
+                // Exclude field that would create indirect circular dependency
+                const indirectCircularDependency = recursiveFindCircular(el);
```

Comment:
You should put that directly in your `if` :wink:

Reply:
Was just to be able to have two comment directly related to the following line but the function name is quite enough self explanatory so I made the suggested change :+1: 
```js
// Exclude field that would create a direct circular dependency
[Some Code]
// Exclude field that would create indirect circular dependency
[Some Code]
```